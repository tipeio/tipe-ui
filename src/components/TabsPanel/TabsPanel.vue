<template>
  <div
    v-if="$slots.default"
    :data-tipe-ui="$options.name"
  >
    <slot/>
  </div>
  <div
    v-else
    :data-tipe-ui="$options.name"
    class="empty"
  >
    empty
  </div>
</template>

<script>
import interfaces from '@/interfaces'

export default {
  name: 'TipeTabsPanel',
  props: {
    ...interfaces.tabsMenuButton
  },
  computed: {
    isActive() {
      console.log('activePanelLabel', this.activePanelLabel)
      return false
    }
  }
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeTabsPanel'] {
  height: 100%;
  width: 100%;
}

.empty {
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 3rem;
  background-color: var(--gray-lightest);
  color: var(--text-gray-light);
  font-size: 0.8125rem;
}
</style>
