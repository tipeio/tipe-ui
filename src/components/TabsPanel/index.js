export { default } from './TabsPanel.vue'
