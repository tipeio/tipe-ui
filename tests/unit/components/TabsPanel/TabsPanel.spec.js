import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import TabsPanel from '@/components/TabsPanel'

describe('TabsMenu', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const renderer = createRenderer()
      const wrapper = shallowMount(TabsPanel)

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })

    it('<root> - has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(TabsPanel)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeTabsPanel')
    })
  })
})
