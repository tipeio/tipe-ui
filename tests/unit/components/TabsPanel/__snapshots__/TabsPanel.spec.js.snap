// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsMenu <template> matches previous snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "element": <div
    class="empty"
    data-tipe-ui="TipeTabsPanel"
  >
    
  empty

  </div>,
  "isFunctionalComponent": undefined,
  "isVueComponent": true,
  "options": Object {
    "attachedToDocument": false,
    "sync": true,
  },
  "version": 2.5,
  "vm": VueComponent {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$el": <div
      class="empty"
      data-tipe-ui="TipeTabsPanel"
    >
      
  empty

    </div>,
    "$emit": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_Ctor": Object {
        "1": [Function],
      },
      "_base": [Function],
      "_parentElm": null,
      "_propKeys": Array [
        "icon",
        "label",
        "active",
      ],
      "_refElm": null,
      "beforeCreate": Array [
        [Function],
      ],
      "components": Object {
        "transition": Object {
          "_Ctor": Object {
            "1": [Function],
          },
          "name": "transition",
          "render": [Function],
        },
        "transition-group": Object {
          "_Ctor": Object {
            "1": [Function],
          },
          "name": "transition-group",
          "render": [Function],
        },
      },
      "computed": Object {
        "isActive": [Function],
      },
      "directives": Object {},
      "filters": Object {},
      "logModifiedComponents": true,
      "methods": undefined,
      "name": "TipeTabsPanel",
      "props": Object {
        "active": Object {
          "default": false,
          "type": [Function],
        },
        "icon": Object {
          "default": "",
          "type": [Function],
        },
        "label": Object {
          "default": "",
          "type": [Function],
        },
      },
      "provide": [Function],
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "__emitted": Object {},
    "__emittedByOrder": Array [],
    "_c": [Function],
    "_computedWatchers": Object {
      "isActive": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": true,
        "expression": "function isActive() {
      console.log('activePanelLabel', this.activePanelLabel);
      return false;
    }",
        "getter": [Function],
        "id": 1,
        "lazy": true,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": true,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
    },
    "_data": Object {},
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": true,
    "_isVue": true,
    "_props": Object {
      "active": false,
      "icon": "",
      "label": "",
    },
    "_provided": undefined,
    "_renderProxy": VueComponent {
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$el": <div
        class="empty"
        data-tipe-ui="TipeTabsPanel"
      >
        
  empty

      </div>,
      "$emit": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_Ctor": Object {
          "1": [Function],
        },
        "_base": [Function],
        "_parentElm": null,
        "_propKeys": Array [
          "icon",
          "label",
          "active",
        ],
        "_refElm": null,
        "beforeCreate": Array [
          [Function],
        ],
        "components": Object {
          "transition": Object {
            "_Ctor": Object {
              "1": [Function],
            },
            "name": "transition",
            "render": [Function],
          },
          "transition-group": Object {
            "_Ctor": Object {
              "1": [Function],
            },
            "name": "transition-group",
            "render": [Function],
          },
        },
        "computed": Object {
          "isActive": [Function],
        },
        "directives": Object {},
        "filters": Object {},
        "logModifiedComponents": true,
        "methods": undefined,
        "name": "TipeTabsPanel",
        "props": Object {
          "active": Object {
            "default": false,
            "type": [Function],
          },
          "icon": Object {
            "default": "",
            "type": [Function],
          },
          "label": Object {
            "default": "",
            "type": [Function],
          },
        },
        "provide": [Function],
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "__emitted": Object {},
      "__emittedByOrder": Array [],
      "_c": [Function],
      "_computedWatchers": Object {
        "isActive": Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": true,
          "expression": "function isActive() {
      console.log('activePanelLabel', this.activePanelLabel);
      return false;
    }",
          "getter": [Function],
          "id": 1,
          "lazy": true,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": true,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      },
      "_data": Object {},
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": true,
      "_isVue": true,
      "_props": Object {
        "active": false,
        "icon": "",
        "label": "",
      },
      "_provided": undefined,
      "_renderProxy": [Circular],
      "_self": [Circular],
      "_staticTrees": null,
      "_uid": 0,
      "_vnode": VNode {
        "asyncFactory": undefined,
        "asyncMeta": undefined,
        "children": Array [
          VNode {
            "asyncFactory": undefined,
            "asyncMeta": undefined,
            "children": undefined,
            "componentInstance": undefined,
            "componentOptions": undefined,
            "context": undefined,
            "data": undefined,
            "elm": 
  empty
,
            "fnContext": undefined,
            "fnOptions": undefined,
            "fnScopeId": undefined,
            "isAsyncPlaceholder": false,
            "isCloned": false,
            "isComment": false,
            "isOnce": false,
            "isRootInsert": false,
            "isStatic": false,
            "key": undefined,
            "ns": undefined,
            "parent": undefined,
            "raw": false,
            "tag": undefined,
            "text": "
  empty
",
          },
        ],
        "componentInstance": undefined,
        "componentOptions": undefined,
        "context": [Circular],
        "data": Object {
          "attrs": Object {
            "data-tipe-ui": "TipeTabsPanel",
          },
          "staticClass": "empty",
        },
        "elm": <div
          class="empty"
          data-tipe-ui="TipeTabsPanel"
        >
          
  empty

        </div>,
        "fnContext": undefined,
        "fnOptions": undefined,
        "fnScopeId": undefined,
        "isAsyncPlaceholder": false,
        "isCloned": false,
        "isComment": false,
        "isOnce": false,
        "isRootInsert": true,
        "isStatic": false,
        "key": undefined,
        "ns": undefined,
        "parent": undefined,
        "raw": false,
        "tag": "div",
        "text": undefined,
      },
      "_watcher": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 2,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": true,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      "_watchers": Array [
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": true,
          "expression": "function isActive() {
      console.log('activePanelLabel', this.activePanelLabel);
      return false;
    }",
          "getter": [Function],
          "id": 1,
          "lazy": true,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": true,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 2,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": true,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      ],
    },
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 0,
    "_vnode": VNode {
      "asyncFactory": undefined,
      "asyncMeta": undefined,
      "children": Array [
        VNode {
          "asyncFactory": undefined,
          "asyncMeta": undefined,
          "children": undefined,
          "componentInstance": undefined,
          "componentOptions": undefined,
          "context": undefined,
          "data": undefined,
          "elm": 
  empty
,
          "fnContext": undefined,
          "fnOptions": undefined,
          "fnScopeId": undefined,
          "isAsyncPlaceholder": false,
          "isCloned": false,
          "isComment": false,
          "isOnce": false,
          "isRootInsert": false,
          "isStatic": false,
          "key": undefined,
          "ns": undefined,
          "parent": undefined,
          "raw": false,
          "tag": undefined,
          "text": "
  empty
",
        },
      ],
      "componentInstance": undefined,
      "componentOptions": undefined,
      "context": [Circular],
      "data": Object {
        "attrs": Object {
          "data-tipe-ui": "TipeTabsPanel",
        },
        "staticClass": "empty",
      },
      "elm": <div
        class="empty"
        data-tipe-ui="TipeTabsPanel"
      >
        
  empty

      </div>,
      "fnContext": undefined,
      "fnOptions": undefined,
      "fnScopeId": undefined,
      "isAsyncPlaceholder": false,
      "isCloned": false,
      "isComment": false,
      "isOnce": false,
      "isRootInsert": true,
      "isStatic": false,
      "key": undefined,
      "ns": undefined,
      "parent": undefined,
      "raw": false,
      "tag": "div",
      "text": undefined,
    },
    "_watcher": Watcher {
      "active": true,
      "cb": [Function],
      "deep": false,
      "depIds": Set {},
      "deps": Array [],
      "dirty": false,
      "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
      "getter": [Function],
      "id": 2,
      "lazy": false,
      "newDepIds": Set {},
      "newDeps": Array [],
      "sync": true,
      "user": false,
      "value": undefined,
      "vm": [Circular],
    },
    "_watchers": Array [
      Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": true,
        "expression": "function isActive() {
      console.log('activePanelLabel', this.activePanelLabel);
      return false;
    }",
        "getter": [Function],
        "id": 1,
        "lazy": true,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": true,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 2,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": true,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
    ],
  },
  "vnode": VNode {
    "asyncFactory": undefined,
    "asyncMeta": undefined,
    "children": Array [
      VNode {
        "asyncFactory": undefined,
        "asyncMeta": undefined,
        "children": undefined,
        "componentInstance": undefined,
        "componentOptions": undefined,
        "context": undefined,
        "data": undefined,
        "elm": 
  empty
,
        "fnContext": undefined,
        "fnOptions": undefined,
        "fnScopeId": undefined,
        "isAsyncPlaceholder": false,
        "isCloned": false,
        "isComment": false,
        "isOnce": false,
        "isRootInsert": false,
        "isStatic": false,
        "key": undefined,
        "ns": undefined,
        "parent": undefined,
        "raw": false,
        "tag": undefined,
        "text": "
  empty
",
      },
    ],
    "componentInstance": undefined,
    "componentOptions": undefined,
    "context": VueComponent {
      "$attrs": Object {},
      "$children": Array [],
      "$createElement": [Function],
      "$el": <div
        class="empty"
        data-tipe-ui="TipeTabsPanel"
      >
        
  empty

      </div>,
      "$emit": [Function],
      "$listeners": Object {},
      "$options": Object {
        "_Ctor": Object {
          "1": [Function],
        },
        "_base": [Function],
        "_parentElm": null,
        "_propKeys": Array [
          "icon",
          "label",
          "active",
        ],
        "_refElm": null,
        "beforeCreate": Array [
          [Function],
        ],
        "components": Object {
          "transition": Object {
            "_Ctor": Object {
              "1": [Function],
            },
            "name": "transition",
            "render": [Function],
          },
          "transition-group": Object {
            "_Ctor": Object {
              "1": [Function],
            },
            "name": "transition-group",
            "render": [Function],
          },
        },
        "computed": Object {
          "isActive": [Function],
        },
        "directives": Object {},
        "filters": Object {},
        "logModifiedComponents": true,
        "methods": undefined,
        "name": "TipeTabsPanel",
        "props": Object {
          "active": Object {
            "default": false,
            "type": [Function],
          },
          "icon": Object {
            "default": "",
            "type": [Function],
          },
          "label": Object {
            "default": "",
            "type": [Function],
          },
        },
        "provide": [Function],
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "$parent": undefined,
      "$refs": Object {},
      "$root": [Circular],
      "$scopedSlots": Object {},
      "$slots": Object {},
      "$vnode": undefined,
      "__emitted": Object {},
      "__emittedByOrder": Array [],
      "_c": [Function],
      "_computedWatchers": Object {
        "isActive": Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": true,
          "expression": "function isActive() {
      console.log('activePanelLabel', this.activePanelLabel);
      return false;
    }",
          "getter": [Function],
          "id": 1,
          "lazy": true,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": true,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      },
      "_data": Object {},
      "_directInactive": false,
      "_events": Object {},
      "_hasHookEvent": false,
      "_inactive": null,
      "_isBeingDestroyed": false,
      "_isDestroyed": false,
      "_isMounted": true,
      "_isVue": true,
      "_props": Object {
        "active": false,
        "icon": "",
        "label": "",
      },
      "_provided": undefined,
      "_renderProxy": VueComponent {
        "$attrs": Object {},
        "$children": Array [],
        "$createElement": [Function],
        "$el": <div
          class="empty"
          data-tipe-ui="TipeTabsPanel"
        >
          
  empty

        </div>,
        "$emit": [Function],
        "$listeners": Object {},
        "$options": Object {
          "_Ctor": Object {
            "1": [Function],
          },
          "_base": [Function],
          "_parentElm": null,
          "_propKeys": Array [
            "icon",
            "label",
            "active",
          ],
          "_refElm": null,
          "beforeCreate": Array [
            [Function],
          ],
          "components": Object {
            "transition": Object {
              "_Ctor": Object {
                "1": [Function],
              },
              "name": "transition",
              "render": [Function],
            },
            "transition-group": Object {
              "_Ctor": Object {
                "1": [Function],
              },
              "name": "transition-group",
              "render": [Function],
            },
          },
          "computed": Object {
            "isActive": [Function],
          },
          "directives": Object {},
          "filters": Object {},
          "logModifiedComponents": true,
          "methods": undefined,
          "name": "TipeTabsPanel",
          "props": Object {
            "active": Object {
              "default": false,
              "type": [Function],
            },
            "icon": Object {
              "default": "",
              "type": [Function],
            },
            "label": Object {
              "default": "",
              "type": [Function],
            },
          },
          "provide": [Function],
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "$parent": undefined,
        "$refs": Object {},
        "$root": [Circular],
        "$scopedSlots": Object {},
        "$slots": Object {},
        "$vnode": undefined,
        "__emitted": Object {},
        "__emittedByOrder": Array [],
        "_c": [Function],
        "_computedWatchers": Object {
          "isActive": Watcher {
            "active": true,
            "cb": [Function],
            "deep": false,
            "depIds": Set {},
            "deps": Array [],
            "dirty": true,
            "expression": "function isActive() {
      console.log('activePanelLabel', this.activePanelLabel);
      return false;
    }",
            "getter": [Function],
            "id": 1,
            "lazy": true,
            "newDepIds": Set {},
            "newDeps": Array [],
            "sync": true,
            "user": false,
            "value": undefined,
            "vm": [Circular],
          },
        },
        "_data": Object {},
        "_directInactive": false,
        "_events": Object {},
        "_hasHookEvent": false,
        "_inactive": null,
        "_isBeingDestroyed": false,
        "_isDestroyed": false,
        "_isMounted": true,
        "_isVue": true,
        "_props": Object {
          "active": false,
          "icon": "",
          "label": "",
        },
        "_provided": undefined,
        "_renderProxy": [Circular],
        "_self": [Circular],
        "_staticTrees": null,
        "_uid": 0,
        "_vnode": [Circular],
        "_watcher": Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 2,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": true,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
        "_watchers": Array [
          Watcher {
            "active": true,
            "cb": [Function],
            "deep": false,
            "depIds": Set {},
            "deps": Array [],
            "dirty": true,
            "expression": "function isActive() {
      console.log('activePanelLabel', this.activePanelLabel);
      return false;
    }",
            "getter": [Function],
            "id": 1,
            "lazy": true,
            "newDepIds": Set {},
            "newDeps": Array [],
            "sync": true,
            "user": false,
            "value": undefined,
            "vm": [Circular],
          },
          Watcher {
            "active": true,
            "cb": [Function],
            "deep": false,
            "depIds": Set {},
            "deps": Array [],
            "dirty": false,
            "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
            "getter": [Function],
            "id": 2,
            "lazy": false,
            "newDepIds": Set {},
            "newDeps": Array [],
            "sync": true,
            "user": false,
            "value": undefined,
            "vm": [Circular],
          },
        ],
      },
      "_self": [Circular],
      "_staticTrees": null,
      "_uid": 0,
      "_vnode": [Circular],
      "_watcher": Watcher {
        "active": true,
        "cb": [Function],
        "deep": false,
        "depIds": Set {},
        "deps": Array [],
        "dirty": false,
        "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
        "getter": [Function],
        "id": 2,
        "lazy": false,
        "newDepIds": Set {},
        "newDeps": Array [],
        "sync": true,
        "user": false,
        "value": undefined,
        "vm": [Circular],
      },
      "_watchers": Array [
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": true,
          "expression": "function isActive() {
      console.log('activePanelLabel', this.activePanelLabel);
      return false;
    }",
          "getter": [Function],
          "id": 1,
          "lazy": true,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": true,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
        Watcher {
          "active": true,
          "cb": [Function],
          "deep": false,
          "depIds": Set {},
          "deps": Array [],
          "dirty": false,
          "expression": "function () {
      vm._update(vm._render(), hydrating);
    }",
          "getter": [Function],
          "id": 2,
          "lazy": false,
          "newDepIds": Set {},
          "newDeps": Array [],
          "sync": true,
          "user": false,
          "value": undefined,
          "vm": [Circular],
        },
      ],
    },
    "data": Object {
      "attrs": Object {
        "data-tipe-ui": "TipeTabsPanel",
      },
      "staticClass": "empty",
    },
    "elm": <div
      class="empty"
      data-tipe-ui="TipeTabsPanel"
    >
      
  empty

    </div>,
    "fnContext": undefined,
    "fnOptions": undefined,
    "fnScopeId": undefined,
    "isAsyncPlaceholder": false,
    "isCloned": false,
    "isComment": false,
    "isOnce": false,
    "isRootInsert": true,
    "isStatic": false,
    "key": undefined,
    "ns": undefined,
    "parent": undefined,
    "raw": false,
    "tag": "div",
    "text": undefined,
  },
}
`;
